File: jsonimport/Source/JsonImport/Private/JsonObjects/JsonSplatPrototype.cpp
#include "JsonImportPrivatePCH.h"
#include "JsonSplatPrototype.h"
#include "macros.h"

//#define JSON_ENABLE_VALUE_LOGGING

void JsonSplatPrototype::load(JsonObjPtr data){
	JSON_GET_VAR(data, textureId);
	JSON_GET_VAR(data, normalMapId);
			
	JSON_GET_VAR(data, metallic);
	JSON_GET_VAR(data, smoothness);
	JSON_GET_VAR(data, specular);
	JSON_GET_VAR(data, tileOffset);
	JSON_GET_VAR(data, tileSize);
}

JsonSplatPrototype::JsonSplatPrototype(JsonObjPtr data){
	load(data);
}

File: ExodusExport/Scripts/Editor/Exporter/JsonSplatPrototype.cs
using UnityEngine;
using UnityEditor;

namespace SceneExport{
	[System.Serializable]
	public class JsonSplatPrototype: IFastJsonValue{
		public SplatPrototype data;
		public ResId normalMapId = ResId.invalid;
		public ResId textureId = ResId.invalid;
		public void writeRawJsonValue(FastJsonWriter writer){
			writer.beginRawObject();
			writer.writeKeyVal("textureId", textureId);
			writer.writeKeyVal("normalMapId", normalMapId);
			
			writer.writeKeyVal("metallic", data.metallic);
			writer.writeKeyVal("smoothness", data.smoothness);
			writer.writeKeyVal("specular", data.specular);
			writer.writeKeyVal("tileOffset", data.tileOffset);
			writer.writeKeyVal("tileSize", data.tileSize);
			writer.endObject();
		}
		
		public JsonSplatPrototype(SplatPrototype data_, ResourceMapper resMap){
			data = data_;
			textureId = resMap.getTextureId(data.texture);
			normalMapId = resMap.getTextureId(data.normalMap);
		}
	}
}